[EuiBasicTable] Show `nameTooltip` on header cell when it's focusable (#8644)
elastic/eui: packages/eui/changelogs/upcoming/8644.md

**Accessibility**

> - Updated `EuiTableHeaderCell` to output the `nameTooltip` for sortable cells on the cell element directly to ensure the tooltip shown on focus

Reviewer: **nit-pick:**

```suggestion
- Updated `EuiTableHeaderCell` to output `nameTooltip` directly on sortable cell elements, ensuring tooltips appear on focus
```
